src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -161,6 +161,13 @@ private static void EmitRegexDerivedTypeRunnerFactory(IndentedTextWriter writer,
             writer.WriteLine($"        // Description:");
             DescribeExpression(writer, rm.Tree.Root.Child(0), "        // ", analysis); // skip implicit root capture
             writer.WriteLine();
+            if (rm.Tree.Culture != null)
+            {
+                // If the RegexTree has Culture set, then we need to emit the textInfo field that should be used at match time for casing operations.
+                writer.WriteLine($"        /// <summary>TextInfo that will be used for Backreference case comparisons.</summary>");
+                writer.WriteLine($"        private readonly TextInfo textInfo = CultureInfo.GetCultureInfo({Literal(rm.Tree.Culture.Name)}).TextInfo;");
```

Comment:
What happens if the specified culture name isn't found?  It doesn't throw an exception, right?

Reply:
This should never happen. The culture is retrieved by using CurrentCulture in construction time, and I have a test that is precisely ensuring that any possible value of that will succeed in the source generation case to produce the right textInfo field:

https://github.com/dotnet/runtime/pull/67184/files#diff-32fe0f9618badcd7c16e85afeac0bdf445ba71c0c51457abb4c10c3ee4787b06R120-R131

Reply:
The code could be running on an entirely different machine. It's still not possible?

Reply:
Chatted offline with @tarekgh about this. TLDR is that this will work in 99.9 percent cases, and will throw exception when it doesn’t.

The longer explanation is that this will always work when using ICU since it has a mode were if you use a culture that isn’t part of ICU, it finds another one that is the closest possible in terms of behavior and use that. When not using ICU and particularly in older Windows versions (like Windows 8) only around 100 cultures are supported (around 12% of all cultures), and when trying to use one of the unsupported ones, CultureInfo will throw an exception at runtime when trying to initialize the culture. We could opt to catch the exception if it happens and fallback to using currentCulture in that case if you think that is better.

Reply:
> We could opt to catch the exception if it happens and fallback to using currentCulture in that case if you think that is better.

The good thing about "99.9%" is it's rare.  The bad thing is there's a good chance this gets deployed and then some machine somewhere some time starts failing, with little recourse but to ship new binaries.  So, yeah, I think we should catch the relevant exception and fall back to CultureInfo.CurrentCulture or CultureInfo.InvariantCulture.

Alternatively, rather than outputting `CultureInfo.GetCultureInfo({Literal(rm.Tree.Culture.Name)})`, we could normalize to just three cultures (one for each of the three groupings we understand), assuming we can pick one for each group that we know will never fail.